src/shared/infra/email/services/email-service.ts

```diff
@@ -0,0 +1,13 @@
+import type { EmailProvider } from '@/shared/infra/email/providers/email-provider.js';
+
+export class EmailService {
+  private emailProvider: EmailProvider;
+
+  constructor(emailProvider: EmailProvider) {
+    this.emailProvider = emailProvider;
+  }
```

Comment:
```suggestion
  constructor(private readonly emailProvider: EmailProvider) { }
```
bom utilizar a syntax sugar do ts

Reply:
done [938c74e]